h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -1913,19 +1913,30 @@ public GLMMojoWriter getMojo() {
     return new GLMMojoWriter(this);
   }
 
-  @Override
-  protected boolean isFeatureUsedInPredict(int featureIdx) {
+  private boolean isFeatureUsedInPredict(int featureIdx, int offset){
     if (featureIdx < _output._dinfo._catOffsets.length - 1 && _output._column_types[featureIdx].equals("Enum")) {
       for (int i = _output._dinfo._catOffsets[featureIdx];
            i < _output._dinfo._catOffsets[featureIdx + 1];
            i++) {
-        if (beta()[i] != 0) return true;
+        if (beta()[i + offset] != 0) return true;
       }
       return false;
     } else {
       featureIdx += _output._dinfo._numOffsets[0] - _output._dinfo._catOffsets.length + 1;
     }
-    return beta()[featureIdx] != 0;
+    return beta()[featureIdx + offset] != 0;
+}
 
+  @Override
+  protected boolean isFeatureUsedInPredict(int featureIdx) {
```

Comment:
is there an easier way of implementing this?

I don't know... for example calling dataInfo.innerProduct on an instance of a dummy row where only feature X is "hot" and the beta coefs?

my concern is that is the is might not be always correct or it might become obsolete, eg. what about interactions - the coefficient for feature could be zero but it might still be used in interactions

Reply:
@michalkurka I understand the concern and I tried to find some other simpler way but I didn't find one. 

I could use the `innerProduct` but since the problem is the transformation from the "feature space" to $R^n$ the way it is written I think I would have to create an array of `Chunk`s (the dummy row) and use specific values to encode categorical variables etc and then use `extract(Dense|Sparse)Row` to do the transformation (so that I don't replicate the logic). 

I also looked at how `score0` works and it seems to work similarly to what I wrote there.

Good point about the interactions - I wasn't able to find a good solution so for now I check if the interactions are set and if they are I will just say all features are used for prediction - better a bit slower than incorrect solution. Note that this scenario is very unlikely since this method is used only in SE and AFAIK to use interactions the user would have to know what names are used in level one frame.

Did I miss some easier solution? Maybe @wendycwong would know?